src/libImaging/GifEncode.c

```diff
@@ -21,136 +22,199 @@
+        /* Hash found experimentally to be pretty good. */
+        /* This works ONLY with TABLE_SIZE a power of 2. */
+        st->probe = ((st->head ^ (st->tail << 6)) * 31) & (TABLE_SIZE - 1);
+        while (st->codes[st->probe]) {
+            if ((st->codes[st->probe] & 0xFFFFF) ==
+                                        ((st->head << 8) | st->tail)) {
+                st->head = st->codes[st->probe] >> 20;
+                goto encode_loop;
             } else {
-                context->flush = context->block;
+        /* Reprobe decrement must be nonzero and relatively prime to table
+         * size. So, any odd positive number for power-of-2 size. */
+                if ((st->probe -= ((st->tail << 2) | 1)) < 0) {
+                    st->probe += TABLE_SIZE;
+                }
             }
         }
-
-        /* get a new block */
-        if (context->free) {
-            block = context->free;
-            context->free = NULL;
+        /* Key not found, probe is at empty slot. */
+        st->code = st->head;
+        st->put_state = PUT_HEAD;
+        goto put_code;
+insert_code_or_clear: /* jump here after put_code */
+        if (st->next_code < CODE_LIMIT) {
+            st->codes[st->probe] = (st->next_code << 20) |
+                                    (st->head << 8) | st->tail;
+            if (st->next_code > st->max_code) {
+                st->max_code = st->max_code * 2 + 1;
+                st->code_width++;
+            }
+            st->next_code++;
         } else {
-            /* malloc check ok, small constant allocation */
-            block = malloc(sizeof(GIFENCODERBLOCK));
-            if (!block) {
-                return 0;
+            st->code = st->clear_code;
+            st->put_state = PUT_CLEAR;
+            goto put_code;
+reset_after_clear: /* jump here after put_code */
+            glzwe_reset(st);
+        }
+        st->head = st->tail;
+        goto encode_loop;
+
+    case LZW_INITIAL:
+        glzwe_reset(st);
+        st->code = st->clear_code;
+        st->put_state = PUT_INIT_CLEAR;
+put_code:
+        st->code_bits_left = st->code_width;
+check_buf_bits:
+        if (!st->buf_bits_left) {   /* out buffer full */
+
+    case LZW_TRY_OUT1:
+            if (!*out_avail) {
+                st->entry_state = LZW_TRY_OUT1;
+                return GLZW_NO_OUTPUT_AVAIL;
             }
+            *out_ptr++ = st->code_buffer;
+            (*out_avail)--;
+            st->code_buffer = 0;
+            st->buf_bits_left = 8;
+        }
+        /* code bits to pack */
+        UINT32 n = st->buf_bits_left < st->code_bits_left
+                        ? st->buf_bits_left : st->code_bits_left;
+        st->code_buffer |=
+                        (st->code & ((1 << n) - 1)) << (8 - st->buf_bits_left);
+        st->code >>= n;
+        st->buf_bits_left -= n;
+        st->code_bits_left -= n;
+        if (st->code_bits_left)
+            goto check_buf_bits;
```

Comment:
```suggestion
        if (st->code_bits_left) {
            goto check_buf_bits;
        }
```
Just a style change, see #4492

Reply:
Done. Thanks for catching that. I tried to conform to the current Pillow style but missed that one.